File: calculadora.py
def suma(a, b):
    return a + b

def resta(a, b):
    return a - b

def multiplicacion(a, b):
    return a * b

def division(a, b):
    if b == 0:
        raise ValueError("No se puede dividir por cero.")
    return a / b

def main():
    print("Calculadora Básica")
    print("Seleccione la operación:")
    print("1. Suma")
    print("2. Resta")
    print("3. Multiplicación")
    print("4. División")

    while True:
        seleccion = input("Ingrese el número de la operación (1/2/3/4): ")

        if seleccion in ['1', '2', '3', '4']:
            num1 = float(input("Ingrese el primer número: "))
            num2 = float(input("Ingrese el segundo número: "))

            if seleccion == '1':
                print(f"Resultado: {suma(num1, num2)}")
            elif seleccion == '2':
                print(f"Resultado: {resta(num1, num2)}")
            elif seleccion == '3':
                print(f"Resultado: {multiplicacion(num1, num2)}")
            elif seleccion == '4':
                try:
                    print(f"Resultado: {division(num1, num2)}")
                except ValueError as e:
                    print(e)
        else:
            print("Selección no válida. Intente de nuevo.")

if __name__ == "__main__":
    main()

File: calculadora.cs
using System;

class Calculadora
{
    static void Main(string[] args)
    {
        Console.WriteLine("Ingrese el primer número:");
        double num1 = Convert.ToDouble(Console.ReadLine());

        Console.WriteLine("Ingrese el segundo número:");
        double num2 = Convert.ToDouble(Console.ReadLine());

        Console.WriteLine("Seleccione una operación:");
        Console.WriteLine("1. Sumar");
        Console.WriteLine("2. Restar");
        Console.WriteLine("3. Multiplicar");
        Console.WriteLine("4. Dividir");
        int operacion = Convert.ToInt32(Console.ReadLine());

        double resultado = 0;

        switch (operacion)
        {
            case 1:
                resultado = num1 + num2;
                break;
            case 2:
                resultado = num1 - num2;
                break;
            case 3:
                resultado = num1 * num2;
                break;
            case 4:
                if (num2 != 0)
                {
                    resultado = num1 / num2;
                }
                else
                {
                    Console.WriteLine("Error: División por cero.");
                    return;
                }
                break;
            default:
                Console.WriteLine("Operación no válida.");
                return;
        }

        Console.WriteLine("El resultado es: " + resultado);
    }
}